### DISCUSSION TITLE
UART basic question

### DISCUSSION THREAD

**Discussion by MrTinkerman** [micropython/micropython#10806] [RP2040 / Pico]
Hi all,

I'm getting started using UART on the Pi Pico and I have some basic questions. I did google but my questions might be to basic and I haven't found any answers yet. I am also not sure if the Pico is the right board for the job because I would need three serial connections and USB debugging in the end.

My progress so far:
- I can control a stepper motor using a TMC2209 stepper driver on UART 1 on the Pico

What am I trying to achieve?
- I would like to separately control two stepper motors using two TMC2209 stepper drivers, these both need a UART connection
- I also want to read serial data from an RS485 line using a converter, this would also need a UART
- I do also really need debuging over USB wile programming

Is this even possible? For example, when I connect my stepper driver on UART 0, Thonny can't make a connection over USB. This might be a 'Thonny quirk' though, I am trying to switch to Pycharm at the moment.

The main question would be: 
Can I use UART0, UART1, a third software UART and USB debugging on the Pico?

If possible I would like to stick with the Pico.
If too complicated, What board would be more suitable in this situation? 

Thanks for reading!

**Comment by 2dof** [comment]
see this topic: https://forums.raspberrypi.com/viewtopic.php?t=333550
or just consider to use some serial port Expander

**Reply by MrTinkerman** [reply]
Thanks for the link, didn't come across that one. Looks like I have some reading up to do on PIO though, this might be the a god excuse to do so.

But I think I'm switching to Arduino for this project, the stepper runs more smoothly that way.

**Comment by robert-hh** [comment]
UART0 and USB are by firmware independent. You can route the USB communication to UART, but that's a software setting.

**Comment by MrTinkerman** [comment]
This is not so much an answer but my decision in this project.

I switched to an Arduino Nano Every. With some modifications this board offers 3 hardware serial ports:
https://forum.arduino.cc/t/arduino-nano-every-access-to-4-serial/614358

I got it up and running so no need  pursuing to do this on a Pico.

Thanks for the input though.
